versioned/transfer-proto/src/main/proto/org/projectnessie/versioned/transfer/serialize/transfer.proto

```diff
@@ -113,3 +125,24 @@ message RepositoryDescriptionProto {
   optional int64 repository_created_timestamp_millis = 4;
   optional string default_branch_name = 5;
 }
+
+enum Encoding {
+  Unknown_Encoding = 0;
+  Smile = 1;
+}
+
+// Custom objects and special standard objects
+message GenericObj {
+  bytes id = 1;
+  // the short name or full name of the (custom) obj type
+  string type_name = 2;
+  bytes data = 3;
+  // UpdateableObj
+  optional string version_token = 4;
+  // custom obj
+  optional Encoding encoding = 5;
+  // the value of `o.p.versioned.storage.common.objtypes.Compression.valueString()`
+  optional string compression = 6;
+  // UniqueIdObj
```

Comment:
`UniqueIdObj` being a standard object type, isn't it better to create a separate proto for it?

Reply:
Renamed the proto type :D

Reply:
Faire enough :-)